File: files/fraser-coho-workshop/semantic-lab/contributions/preview-term-requests.py
"""Local candidate/text preview; run from the workshop root after preparation."""
from pathlib import Path
import sys
import pandas as pd
import metasalmonpy as ms

sys.path.insert(0, str(Path("scripts").resolve()))
from workshop import check_preparation

if ms.__version__ != "0.4.0":
    raise RuntimeError("Use the workshop's pinned metasalmonpy 0.4.0 release.")
check_preparation()
destination = Path("output") / "term-request-preview-python"
if destination.exists():
    raise FileExistsError("Preview folder exists; preserve it and choose a new destination in this script.")
suggestions = pd.read_csv("checkpoints/seeded-sdp/semantic_suggestions.csv")
gaps = ms.detect_semantic_term_gaps(suggestions=suggestions)
previews = ms.render_ontology_term_request(
    gaps, scope="auto", ask=False,
    profile_name="fraser-coho-workshop-draft",
)
destination.mkdir(parents=True)
gaps.to_csv(destination / "candidate-gaps.csv", index=False)
for i, row in enumerate(previews.itertuples(index=False), start=1):
    (destination / f"request-{i:02d}.md").write_text(
        "# SOFTWARE PREVIEW: not a reviewed gap or submitted request\n\n"
        + row.request_title + "\n\n" + row.request_body + "\n",
        encoding="utf-8",
    )
print(f"{len(gaps)} candidate rows; {len(previews)} local request previews. Review against your source evidence.")

File: files/fraser-coho-workshop/semantic-lab/contributions/preview-term-requests.R
# Run from the extracted workshop root after actual Chapter 2-3 preparation.
# Candidate detection and text rendering only: no retrieval, AI, or submission.
stopifnot(as.character(utils::packageVersion("metasalmon")) == "0.5.0")
source("scripts/workshop.R")
check_preparation()
destination <- file.path("output", "term-request-preview-r")
if (dir.exists(destination)) stop("Preview folder exists; preserve it and choose a new destination in this script.")
suggestions <- readr::read_csv(
  "checkpoints/seeded-sdp/semantic_suggestions.csv", show_col_types = FALSE
)
gaps <- metasalmon::detect_semantic_term_gaps(suggestions = suggestions)
previews <- metasalmon::render_ontology_term_request(
  gaps, scope = "auto", ask = FALSE,
  profile_name = "fraser-coho-workshop-draft"
)
dir.create(destination, recursive = TRUE)
readr::write_csv(gaps, file.path(destination, "candidate-gaps.csv"))
if (nrow(previews)) {
  for (i in seq_len(nrow(previews))) {
    writeLines(c("# SOFTWARE PREVIEW: not a reviewed gap or submitted request", "",
      previews$request_title[[i]], "", previews$request_body[[i]]),
      file.path(destination, sprintf("request-%02d.md", i)))
  }
}
message(nrow(gaps), " candidate rows; ", nrow(previews), " local request previews. Review against your source evidence.")
